frontend/src/models/connections.ts

```diff
@@ -0,0 +1,160 @@
+import { createModel } from '@rematch/core'
+import { API_URL, DEVELOPER_KEY } from '../shared/constants'
+import { newConnection, setConnection } from '../helpers/connectionHelper'
+import { r3, getToken } from '../services/remote.it'
+import { selectById } from '../models/devices'
+import { RootModel } from './rootModel'
+import { store } from '../store'
+import axios from 'axios'
+
+type ConnectionsState = {
+  all: IConnection[]
+}
+
+const state: ConnectionsState = {
+  all: [],
+}
+
+export default createModel<RootModel>()({
+  state,
+  effects: dispatch => ({
+    async updateConnection(connection: IConnection, globalState) {
+      globalState.connections.all.some((c, index) => {
+        if (c.id === connection.id) {
+          globalState.connections[index] = connection
+          dispatch.backend.set({ connections: globalState.connections.all })
+          if (connection) return true
+        }
+        return false
+      })
+    },
+
+    async restoreConnections(connections: IConnection[], globalState) {
+      connections.forEach(async connection => {
+        // data missing from cli if our connections file is lost
+        if (!connection.owner) {
+          const [service] = selectById(globalState, connection.id)
+          if (service) {
+            connection = { ...newConnection(service), ...connection }
+            setConnection(connection)
+          } else {
+            console.warn('No service found for connection', connection.id)
```

Comment:
do you need that?

Reply:
Yes - this will help for troubleshooting